File: py/examples/03_chat_ai.py
import os
from imbrace import ImbraceClient

client = ImbraceClient(
    access_token=os.environ["IMBRACE_ACCESS_TOKEN"],
    organization_id=os.environ.get("IMBRACE_ORG_ID"),
    env="stable",
)

# List available models
models = client.chat_ai.list_models()
print("Models:", [m["id"] for m in models.get("data", [])])

# Chat completion
response = client.chat_ai.chat({
    "model": "gpt-4o",
    "messages": [
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Summarize iMBRACE in one sentence."},
    ],
})
print("AI:", response["choices"][0]["message"]["content"])

# Manage chats
chats = client.chat_ai.list_chats()
print("Existing chats:", len(chats))

new_chat = client.chat_ai.create_chat({"title": "SDK Demo Chat"})
print("Created chat:", new_chat.get("id"))

# Manage knowledge bases
knowledge_list = client.chat_ai.list_knowledge()
print("Knowledge bases:", len(knowledge_list))

client.close()

File: ts/examples/03_chat_ai.ts
import { ImbraceClient } from "../src/index.js"

const client = new ImbraceClient({
  accessToken: process.env.IMBRACE_ACCESS_TOKEN!,
  organizationId: process.env.IMBRACE_ORG_ID,
  env: "stable",
})

// List available models
const { data: models } = await client.chatAi.listModels()
console.log("Models:", models.map(m => m.id))

// Chat completion
const response = await client.chatAi.chat({
  model: "gpt-4o",
  messages: [
    { role: "system", content: "You are a helpful assistant." },
    { role: "user", content: "Summarize iMBRACE in one sentence." },
  ],
})
console.log("AI:", response.choices[0].message.content)

// Manage chats
const chats = await client.chatAi.listChats()
console.log("Existing chats:", chats.length)

const newChat = await client.chatAi.createChat({ title: "SDK Demo Chat" })
console.log("Created:", newChat.id)

// Manage knowledge bases
const knowledge = await client.chatAi.listKnowledge()
console.log("Knowledge bases:", knowledge.length)
